datasets/codah/codah.py

```diff
@@ -0,0 +1,140 @@
+# limitations under the License.
+"""The COmmonsense Dataset Adversarially-authored by Humans (CODAH)"""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{chen2019codah,
+  title={CODAH: An Adversarially-Authored Question Answering Dataset for Common Sense},
+  author={Chen, Michael and D'Arcy, Mike and Liu, Alisa and Fernandez, Jared and Downey, Doug},
+  booktitle={Proceedings of the 3rd Workshop on Evaluating Vector Space Representations for NLP},
+  pages={63--69},
+  year={2019}
+}
+"""
+
+_DESCRIPTION = """\
+The COmmonsense Dataset Adversarially-authored by Humans (CODAH) is an evaluation set for commonsense \
+question-answering in the sentence completion style of SWAG. As opposed to other automatically \
+generated NLI datasets, CODAH is adversarially constructed by humans who can view feedback \
+from a pre-trained model and use this information to design challenging commonsense questions. \
+Our experimental results show that CODAH questions present a complementary extension to the SWAG dataset, testing additional modes of common sense.
+"""
+
+_URL = "https://raw.githubusercontent.com/Websail-NU/CODAH/master/data/"
+_FULL_DATA_URL = _URL + "full_data.tsv"
+
+QUESTION_CATEGORIES_MAPPING = {
+    "i": "Idioms",
+    "r": "Reference",
+    "p": "Polysemy",
+    "n": "Negation",
+    "q": "Quantitative",
+    "o": "Others",
+    "": "",
+}
+
+
+class CodahConfig(datasets.BuilderConfig):
+    """BuilderConfig for CODAH."""
+
+    def __init__(self, fold=None, **kwargs):
+        """BuilderConfig for CODAH.
+
+        Args:
+          fold: `string`, official cross validation fold.
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(CodahConfig, self).__init__(**kwargs)
+        self.fold = fold
+
+
+class Codah(datasets.GeneratorBasedBuilder):
+    """The COmmonsense Dataset Adversarially-authored by Humans (CODAH)"""
+
+    VERSION = datasets.Version("1.0.0")
+    BUILDER_CONFIGS = [
+        CodahConfig(name="codah", version=datasets.Version("1.0.0"), description="Full CODAH dataset", fold=None),
+        CodahConfig(
+            name="fold_0", version=datasets.Version("1.0.0"), description="Official CV split (fold_0)", fold="fold_0"
+        ),
+        CodahConfig(
+            name="fold_1", version=datasets.Version("1.0.0"), description="Official CV split (fold_1)", fold="fold_1"
+        ),
+        CodahConfig(
+            name="fold_2", version=datasets.Version("1.0.0"), description="Official CV split (fold_2)", fold="fold_2"
+        ),
+        CodahConfig(
+            name="fold_3", version=datasets.Version("1.0.0"), description="Official CV split (fold_3)", fold="fold_3"
+        ),
+        CodahConfig(
+            name="fold_4", version=datasets.Version("1.0.0"), description="Official CV split (fold_4)", fold="fold_4"
+        ),
```

Comment:
The dataset contains official CV folds, included those here along with full dataset